File: Highest Altitude/HighestAltitude.java
class Solution {
    public int largestAltitude(int[] gain){
        int current = 0;  // Start at altitude 0
        int highest = 0;  // Highest altitude so far

        for (int g : gain){
            current += g; // Update current altitude
            highest = Math.max(highest, current); // Update highest if needed
        }

        return highest;
    }

    public static void main (String[] args){
        Solution sol = new Solution();
        int[] gain1 = {-5,1,5,0,-7};
        System.out.println(sol.largestAltitude(gain1));

        int[] gain2 = {-4,-3,-2,-1,4,3,2};
        System.out.println(sol.largestAltitude(gain2));
    }
}

File: Highest Altitude/highest_altitude.py
from typing import List

class Solution:
    def largestAltitude(self, gain: List[int]) -> int:
        altitude = 0
        max_altitude = 0

        for change in gain:
            altitude += change  # Update current altitude
            max_altitude = max(max_altitude, altitude)  # Track highest

        return max_altitude

if __name__ == "__main__":
    sol = Solution()

    gain1 = [-5,1,5,0,-7]
    print("Output 1: ", sol.largestAltitude(gain1))

    gain2 = [-4,-3,-2,-1,4,3,2]
    print("Output 2: ", sol.largestAltitude(gain2))